scripts/plugins/install-supp-pack

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+
+import subprocess
+
+# Simple XenAPI plugin
+import XenAPIPlugin, time
+
+class PluginError(Exception):
+    """Base Exception class for all plugin errors."""
+    def __init__(self, *args):
+        Exception.__init__(self, *args)
+
+class ArgumentError(PluginError):
+    """Raised when the arguments to the plugin are invalid"""
+
+    def __init__(self, *args):
+        PluginError.__init__(self, *args)
+
+class SubprocessFailure(PluginError):
+    """Raised when there was a failure attaching the VDI"""
+
+    def __init__(self, *args):
+        PluginError.__init__(self, *args)
+
+def install(session, args):
+    if not args.has_key("vdi"):
+        raise ArgumentError("Key 'vdi' missing from arguments")
+    vdi=args['vdi']
+    process = subprocess.Popen([
+        "/opt/xensource/debug/with-vdi %s /opt/xensource/bin/xe-install-supplemental-pack" % vdi],
+        shell = True,
```

Comment:
According to the Python docs, this is a security hole. See the big red warning section here: https://docs.python.org/2/library/subprocess.html#frequently-used-arguments